.github/workflows/spark_test.yaml

```diff
@@ -7,8 +7,10 @@ jobs:
       matrix:
         # These Scala versions must match those in the build.sbt
         scala: [2.12.18, 2.13.13]
+        shard: [0, 1, 2]
     env:
       SCALA_VERSION: ${{ matrix.scala }}
+      NUM_SHARDS: 3
```

Comment:
https://github.com/delta-io/delta/pull/3517/files#diff-b77047dc65c62d814f7f67eec57c23c4cf9d9796e6a8a018558e0e935739d8d8R10 

The Github matrix strategies itself. Github CI itself will read the matrix fields, and create 1 job per matrix combination. It's important NUM_SHARDS == length (matrix.shards) I've commented that in the Yaml so anyone who tries to change this will be super aware of the assumptions, and I also commented how the sharding logic works in TestParallelization.

We just need the environment variable NUM_SHARDS so that in TestParallelization we can determine if a given test is assigned to a shard based on % NUM_SHARDS == the assigned shard ID.

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-a-matrix-for-your-jobs#about-matrix-strategies